modules/dhcp_common/server.rb

```diff
@@ -173,7 +173,10 @@ def clean_up_add_record_parameters(in_options)
       to_return.delete("subnet") # Not a valid key; remove it to prevent conflict with :subnet
       subnet = to_return.delete("network")
 
-      [name || hostname, ip, mac, subnet, to_return.merge!(:hostname => hostname || name)]
+      deleteable = to_return.delete("deleteable")
```

Comment:
I think a better approach would be to initialize ::Proxy::DHCP::Reservation with options hash containing ```:deleteable => true```.